git-link.el

```diff
@@ -285,9 +286,14 @@ return (FILENAME . REVISION) otherwise nil."
       (cond
        ((eq major-mode 'dired-mode)
         (setq filename (dired-file-name-at-point)))
-       ((and (string-match-p "^magit-" (symbol-name major-mode))
-             (fboundp 'magit-file-at-point))
-        (setq filename (magit-file-at-point)))))
+       ((or
+         (git-link--using-magit-blame)
+         (string-match-p "^magit-" (symbol-name major-mode)))
+        (setq filename (or
+                        (and (fboundp 'magit-file-at-point)
+                             (magit-file-at-point))
+                        (and (fboundp 'magit-current-file)
+                             (magit-current-file)))))))
```

Comment:
And in a `^magit-` major mode this will always be bound  but will return `nil` unless we're in blame mode.

Reply:
Maybe the: 


```el
(or (git-link--using-magit-blame)
        (string-match-p "^magit-" (symbol-name major-mode)))
  (or (and (fboundp 'magit-file-at-point)
   ....
```
code makes sense if it's being used as a catchall for other  `^magit-` modes so that we make a best effort attempt to get a filename.